apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/v4/define/ControllerForLowVersionInstrumentation.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.spring.mvc.v4.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+
+import static net.bytebuddy.matcher.ElementMatchers.any;
+
+public class ControllerForLowVersionInstrumentation extends AbstractControllerInstrumentation {
+    public static final String WITNESS_CLASSES_LOW_VERSION = "org.springframework.web.context.support.ServletContextPropertyPlaceholderConfigurer";
+
+    public static final String ENHANCE_ANNOTATION = "org.springframework.stereotype.Controller";
+
+    @Override
+    public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
+        ConstructorInterceptPoint constructorInterceptPoint = new ConstructorInterceptPoint() {
+            @Override
+            public ElementMatcher<MethodDescription> getConstructorMatcher() {
+                return any();
+            }
+
+            @Override
+            public String getConstructorInterceptor() {
+                return "org.apache.skywalking.apm.plugin.spring.mvc.v4.ControllerForLowVersionConstructorInterceptor";
+            }
+        };
+        return new ConstructorInterceptPoint[]{constructorInterceptPoint};
+    }
+
+    @Override
+    protected String[] witnessClasses() {
+        return new String[]{WITHNESS_CLASSES, "org.springframework.cache.interceptor.DefaultKeyGenerator", WITNESS_CLASSES_LOW_VERSION};
```

Comment:
@SummerOfServenteen 
This issue only happened before the version 4.2.0. The `path` method is only supported after version 4.2.0. 

Why does the `org.springframework.cache.interceptor.DefaultKeyGenerator` class add the witness class list? I found that the class also in version 4.2.0

I think that the `org.springframework.web.context.support.ServletContextPropertyPlaceholderConfigurer` class as the witness class will be ok. because of the class removed after version 4.2.0.

Reply:
They are retained to distinguish between V3 and V5

Reply:
The witness class means that the Instrumentation will be active when the witness class is available.  and `ServletContextPropertyPlaceholderConfigurer` is available before version 4.2.0, and it means that`ControllerForLowVersionInstrumentation` will be active before version 4.2.0.  So I am confused that why need to distinguish between V3 and V5？

Reply:
@SummerOfServenteen If the target class is not there in v3, then you don't witness class.